hooks/useEstimateGas.ts

```diff
@@ -1,22 +1,40 @@
 import { useCallback, useMemo } from 'react';
+import { encodeFunctionData, type EstimateContractGasParameters } from 'viem';
 import { usePublicClient } from 'wagmi';
+import { optimism } from 'wagmi/chains';
 import { useWeb3 } from './useWeb3';
 import { isE2E } from 'utils/client';
 import { getAlchemyProvider } from 'utils/providers';
-import { EstimateGasParameters } from 'viem';
+import { l1GasPriceOracleABI } from 'types/abi';
 
 export default function useEstimateGas() {
   const { chain } = useWeb3();
   const e2ePublicClient = usePublicClient();
   const publicClient = useMemo(() => (isE2E ? e2ePublicClient : getAlchemyProvider(chain)), [chain, e2ePublicClient]);
 
   return useCallback(
-    async (request: EstimateGasParameters) => {
+    async (request?: EstimateContractGasParameters) => {
       if (!publicClient) return;
 
+      let l1Gas = 0n;
+      if (chain.id === optimism.id) {
+        if (request) {
+          const data = encodeFunctionData(request);
+          l1Gas = await publicClient.readContract({
+            abi: l1GasPriceOracleABI,
+            address: '0x420000000000000000000000000000000000000F',
+            functionName: 'getL1Fee',
+            args: [data],
+          });
+        } else {
+          l1Gas = 5_555_555_555_55n;
```

Comment:
where does this come from?

Reply:
the thousand separators are shifted

Reply:
The value is only used in case we don't have a request ready to do a gas estimation (for example, on initial render of the modal we don't have an input, so that's a missing argument). I ran some experiments and just ended up choosing this as a magic number.